### PR TITLE
EIP-7516: BLOBBASEFEE opcode

### PR DESCRIPTION
**Pull request by rakita**
Cancun: https://eips.ethereum.org/EIPS/eip-7516
add Blobfee opcode

### CODE REVIEW

**Comment by rakita** [comment]
Waiting for on possible naming change: https://github.com/ethereum/go-ethereum/pull/28098/files#r1327222175
But it is ready to be merged.

File: crates/interpreter/src/instructions/host_env.rs [hunk 1]
```diff
@@ -53,6 +53,16 @@ pub fn basefee<SPEC: Spec>(interpreter: &mut Interpreter, host: &mut dyn Host) {
     push!(interpreter, host.env().block.basefee);
 }
 
+pub fn blobfee<SPEC: Spec>(interpreter: &mut Interpreter, host: &mut dyn Host) {
+    gas!(interpreter, gas::BASE);
+    // EIP-7516: BLOBBASEFEE opcode
+    check!(interpreter, SPEC::enabled(CANCUN));
```
Comment (DaniPopes) [inline crates/interpreter/src/instructions/host_env.rs:59]: check before?

```suggestion
    // EIP-7516: BLOBBASEFEE opcode
    check!(interpreter, SPEC::enabled(CANCUN));
    gas!(interpreter, gas::BASE);
```

**Review by DaniPopes — APPROVED** [review]

File: crates/interpreter/src/instructions/host_env.rs (hunk 1 above)
Comment (rakita) [reply crates/interpreter/src/instructions/host_env.rs]: Yep, that is nicer order

File: crates/primitives/src/env.rs [hunk 2]
```diff
@@ -45,24 +43,62 @@ pub struct BlockEnv {
...
+        }
+    }
 }
 
 impl BlockEnv {
+    /// Takes `blob_excess_gas` saves it inside env
+    /// and calculates `blob_fee` with [`BlobGasAndFee`].
+    pub fn set_blob_gas_and_fee(&mut self, excess_blob_gas: u64) {
+        self.blob_gas_and_fee = Some(BlobGasAndFee::new(excess_blob_gas));
+    }
```
Comment (Rjected) [inline crates/primitives/src/env.rs:83]: I like this, makes it very easy to set both when initializing the env

File: crates/interpreter/src/instructions/opcode.rs [hunk 3]
```diff
@@ -264,7 +265,7 @@ pub const OPCODE_JUMPMAP: [Option<&'static str>; 256] = [
     /* 0x47 */ Some("SELFBALANCE"),
     /* 0x48 */ Some("BASEFEE"),
     /* 0x49 */ Some("BLOBHASH"),
-    /* 0x4a */ None,
+    /* 0x4a */ Some("BLOBFEE"),
```
Comment (Rjected) [inline crates/interpreter/src/instructions/opcode.rs:268]: does it matter that the opcode is called `BLOBBASEFEE` in the spec here?

**Review by Rjected — APPROVED** [review]
LGTM, just one question about the name

File: crates/interpreter/src/instructions/opcode.rs (hunk 3 above)
Comment (rakita) [reply crates/interpreter/src/instructions/opcode.rs:268]: This: https://github.com/bluealloy/revm/pull/721#issuecomment-1721200005
I like BLOBFEE more :)

Comment (Rjected) [reply crates/interpreter/src/instructions/opcode.rs:268]: ah I missed that, yeah I like `BLOBFEE` more too

Comment (rakita) [reply crates/interpreter/src/instructions/opcode.rs]: Renamed back to `BLOBBASEFEE` per Eth R&D discussion

**Comment by rakita** [comment]
@Rjected can you review, had some small (name) changes, it is ready to be merged

**Review by Rjected — APPROVED** [review]
LGTM!

**Pull request merged**